src/Core/src/Platform/Windows/ContentPanel.cs

```diff
@@ -97,18 +97,32 @@ public void UpdateBackground(Paint? background)
 			_borderPath.UpdateBackground(background);
 		}
 
+		[Obsolete("Use Microsoft.Maui.Platform.UpdateBorderStroke instead")]
 		public void UpdateBorderShape(IShape borderShape)
 		{
-			_borderShape = borderShape;
+			if (borderShape is null || _borderPath is null)
+				return;
+
+			_borderPath.UpdateBorderShape(borderShape, ActualWidth, ActualHeight);
+			UpdateContent();
+			UpdateClip(borderShape);
+		}
+		
+		internal void UpdateBorderStroke(IBorderStroke borderStroke)
```

Comment:
I think this can be public on net8 no @hartez ? or else the Obsolete comment will not make sense since this is internal users can't call it